### DISCUSSION TITLE
Is Sidekiq::Component in Sidekiq 6.5+ considered pubilc or private?

### DISCUSSION THREAD

**Discussion by wspurgin** [sidekiq/sidekiq#5389] [Q&A] [answered]
Maybe a question more for the maintainers, but with Sidekiq 6.5, `Sidekiq::Utils` was removed. As called out in https://github.com/mperham/sidekiq/issues/5372, that module was technically private/internal and not meant to be used by others outside the Sidekiq core library.

I confess though, that I used it for its nice-to-haves like `#safe_thread`, `#identity`, etc. when creating threads for activities based on lifecycle events.. (e.g. `on(:startup) { something_including_sidekiq_utils.start }`).

My first thought, upon looking at the changes in 6.5 was "wait, `Sidekiq::Util` was a private API? Bummer" (I missed the ["not intended for extension"](https://github.com/mperham/sidekiq/blob/v6.4.2/lib/sidekiq/util.rb#L8-L11) in the module doc). However there appears to be the new [`Sidekiq::Component`](https://github.com/mperham/sidekiq/blob/v6.5.0/lib/sidekiq/component.rb#L2-L3) which has all the things I was using `Sidekiq::Util` for....

Is it going to be a public "core" API though? If not, then it's probably not wise for me to use it

**Comment by mperham** [comment, marked as answer]
Today lifecycle events and server-side middleware are the major APIs documented for component developers.

Yes, my intent is for `Sidekiq::Component` to be public API for extensions running within a Sidekiq process. I'm not fully clear yet on what that will look like so I haven't made any commitments on long-term support. I generally prototype this stuff with Sidekiq Pro and Enterprise's features but since this will enable a new set of features, there's still some unknowns TBD.

**Reply by wspurgin** [reply]
Thanks @mperham :bow: Appreciate your quick response. I'll just stay plugged in as `Sidekiq::Component` gets built out. If there's any feedback or forum you'd like those of us in the community (free, Sidekiq Pro / Sidekiq Enterprise) around this specific topic, I'd be happy to help however I can.
